lib/platform/gitlab/index.ts

```diff
@@ -442,33 +436,35 @@ export async function getPrList(): Promise<Pr[]> {
 }
 
 async function tryPrAutomerge(
-  pr: number,
+  iid: number,
   platformOptions: PlatformPrOptions
 ): Promise<void> {
   if (platformOptions?.gitLabAutomerge) {
     try {
       const desiredStatus = 'can_be_merged';
       const retryTimes = 5;
 
+      let mr: GitLabMergeRequest;
       // Check for correct merge request status before setting `merge_when_pipeline_succeeds` to  `true`.
       for (let attempt = 1; attempt <= retryTimes; attempt += 1) {
-        const { body } = await gitlabApi.getJson<{
-          merge_status: string;
-          pipeline: string;
-        }>(`projects/${config.repository}/merge_requests/${pr}`);
+        const { body } = await gitlabApi.getJson<GitLabMergeRequest>(
+          `projects/${config.repository}/merge_requests/${iid}`
+        );
         // Only continue if the merge request can be merged and has a pipeline.
         if (body.merge_status === desiredStatus && body.pipeline !== null) {
+          mr = body;
           break;
         }
         await delay(500 * attempt);
       }
 
       await gitlabApi.putJson(
-        `projects/${config.repository}/merge_requests/${pr}/merge`,
+        `projects/${config.repository}/merge_requests/${iid}/merge`,
         {
           body: {
             should_remove_source_branch: true,
             merge_when_pipeline_succeeds: true,
+            squash: mr.squash,
```

Comment:
What happens to older gitlab versions? will thy reject or ignore that property?

Reply:
not sure how we can test that. 

It seems like gitlab doesn't mind if you add extra data to the URL query string or body:

[Example](https://gitlab.com/api/v4/projects/inkscape%2finkscape/merge_requests/1?randomQs=123)